File: Shubham Garad/Microsoft/Bulls&Cows.cpp
class Solution {
public:
    string getHint(string secret, string guess) {
        
        int bulls = 0 , cow = 0;

        unordered_map<char,int>mp1 , mp2;

        for(int i = 0 ;i< secret.size();++i)
        {
            if(secret[i]==guess[i])
            {
                bulls++;
            }
            else
            {
                mp1[secret[i]]++;
                mp2[guess[i]]++;
            }
        }
        for(auto it : mp1)
        {
            if(mp2.find(it.first) != mp2.end())
            {
                cow += min(it.second , mp2[it.first]);
            }
        }
        string ans = to_string(bulls)+"A"+to_string(cow)+"B";
        return ans;
    }
};

File: RAO.sh/Microsoft/Day I/3 Bulls and Cows/Bulls&Cows.go
package main

import "strconv"

func getHint(secret string, guess string) string {
	bulls, cows := 0, 0
	freq := make(map[int]int)

	for i := 0; i < len(secret); i++ {
		if secret[i] == guess[i] {
			bulls++
		} else {
			if freq[int(secret[i])] < 0 {
				cows++
			}
			freq[int(secret[i])]++

			if freq[int(guess[i])] > 0 {
				cows++
			}
			freq[int(guess[i])]--
		}
	}

	return strconv.Itoa(bulls) + "A" + strconv.Itoa(cows) + "B"
}

func main() {
	secret := "1807"
	guess := "7810"

	println(getHint(secret, guess))
}
